### ISSUE TITLE
[JENKINS-43897] Some node dependency requires (it's a mystery which one)

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#4065]
<p>if you try to build blue ocean from master on windows with out python on your path you receive a (un)helpful error</p>

<p><tt><span class="error">&#91;ERROR&#93;</span> gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.</tt></p>

<p>if you set that variable and build then try again you get...</p>

<p><tt><span class="error">&#91;ERROR&#93;</span> gyp ERR! stack Error: Can't find Python executable "c:\Java\python-3.6.1\python.exe ", you can set the PYTHON env variable.</tt></p>

<p>So the build is just plain lying. (also note the extra space after the variable)</p>

```
C:\blueocean-plugin>echo %PYTHON%
c:\Java\python-3.6.1\python.exe

C:\workarea\source\github\jenkinsci\blueocean-plugin>echo THERE_ARE_NO_SPACES___%PYTHON%__
THERE_ARE_NO_SPACES___c:\Java\python-3.6.1\python.exe__

C:\blueocean-plugin>file %PYTHON%
c:\Java\python-3.6.1\python.exe: PE32+ executable (console) x86-64, for MS Windows

```
<p>(BTW it would be nice to know ahead of time that you need to install tools other than Maven to build the thing and what the versions required are. only after looking at the js file did I see that it requires a python version < 3 - which would save some time finding and installing the package...)</p>

<p> </p>

<p>(<a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=teilo" class="user-hover" rel="teilo">teilo</a> can you past in more of the log? above isn't useful)</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/teilo.png" title="teilo's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=teilo">teilo</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43897" target="_blank">Some node dependency requires (it's a mystery which one)</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers">cliffmeyers</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: blueocean-plugin
<li><b>resolution</b>: Won't Do
<li><b>resolved</b>: 2017-07-25T00:09:00+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>if you try to build blue ocean from master on windows with out python on your path you receive a (un)helpful error</p>

<p><tt><span class="error">&#91;ERROR&#93;</span> gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.</tt></p>

<p>if you set that variable and build then try again you get...</p>

<p><tt><span class="error">&#91;ERROR&#93;</span> gyp ERR! stack Error: Can't find Python executable "c:\Java\python-3.6.1\python.exe ", you can set the PYTHON env variable.</tt></p>

<p>So the build is just plain lying. (also note the extra space after the variable)</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>C:\blueocean-plugin&gt;echo %PYTHON%
c:\Java\python-3.6.1\python.exe

C:\workarea\source\github\jenkinsci\blueocean-plugin&gt;echo THERE_ARE_NO_SPACES___%PYTHON%__
THERE_ARE_NO_SPACES___c:\Java\python-3.6.1\python.exe__

C:\blueocean-plugin&gt;file %PYTHON%
c:\Java\python-3.6.1\python.exe: PE32+ executable (console) x86-64, for MS Windows
</pre>
</div></div>
<p>(BTW it would be nice to know ahead of time that you need to install tools other than Maven to build the thing and what the versions required are. only after looking at the js file did I see that it requires a python version &lt; 3 - which would save some time finding and installing the package...)</p>

<p> </p>

<p>(<a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=teilo" class="user-hover" rel="teilo">teilo</a> can you past in more of the log? above isn't useful)</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Windows...</code></li></ul>
<details><summary><i>2 attachments</i></summary>

- [no_python.txt](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/37482/no_python.txt)
- [with_python.txt](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/37483/with_python.txt)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43897?focusedId=297517&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297517">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>solution: if we could not use npm, that would be nice. There is not really anything we can do about what npm dependencies try to build on different platforms unfortunately, no way to know ahead of time that some stupid thing thinks it needs python. </p>

<p>It would help if there was more log about what npm dependency it was trying to install when it needs it (I don't think anyone knew python was needed for something to be installed, there is nothing in blue ocean directly obviously that needs it). </p>

<p> </p>

<p>Some stupid npm dependencies need stupid compilers because... who knows why. </p>
</pre>
</details>

</ul>
</details>

<p>solution: if we could not use npm, that would be nice. There is not really anything we can do about what npm dependencies try to build on different platforms unfortunately, no way to know ahead of time that some stupid thing thinks it needs python. </p>

<p>It would help if there was more log about what npm dependency it was trying to install when it needs it (I don't think anyone knew python was needed for something to be installed, there is nothing in blue ocean directly obviously that needs it). </p>

<p> </p>

<p>Some stupid npm dependencies need stupid compilers because... who knows why. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43897?focusedId=297518&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297518">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>"only after looking at the js file" - what js file was that? </p>
</pre>
</details>

</ul>
</details>

<p>"only after looking at the js file" - what js file was that? </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43897?focusedId=297521&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297521">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>see also: <a href="https://www.npmjs.com/package/node-gyp" class="external-link" target="_blank" rel="nofollow noopener">https://www.npmjs.com/package/node-gyp</a> (which probably should be in the README)</p>

<p> </p>

<p>PR to have it in the readme: <a href="https://github.com/jenkinsci/blueocean-plugin/pull/1012" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/1012</a></p>

<p>It would be nice to know what npm module is doing this (as I assume it wasn't before) but there isn't any information in the log above about that. Ideally those modules can then be avoided. </p>

<p> </p>

<p>The problem is it can happen just due to a transitive dependency from a version upgrade (routine change) - not in blue ocean code or dependencies directly. </p>

<p> </p>
</pre>
</details>

</ul>
</details>

<p>see also: <a href="https://www.npmjs.com/package/node-gyp" class="external-link" target="_blank" rel="nofollow noopener">https://www.npmjs.com/package/node-gyp</a> (which probably should be in the README)</p>

<p> </p>

<p>PR to have it in the readme: <a href="https://github.com/jenkinsci/blueocean-plugin/pull/1012" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/1012</a></p>

<p>It would be nice to know what npm module is doing this (as I assume it wasn't before) but there isn't any information in the log above about that. Ideally those modules can then be avoided. </p>

<p> </p>

<p>The problem is it can happen just due to a transitive dependency from a version upgrade (routine change) - not in blue ocean code or dependencies directly. </p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/teilo.png" title="teilo's avatar" /> teilo:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43897?focusedId=297559&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297559">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I think it is all bufferutil, but will clean everything and run again....</p>

<p>which gets you to the bit that then requires MSBuild... (all to bundle and process some javascript :-o  )</p>

<p>at which point I shrug and walk away <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>will post logs shortly...</p>
</pre>
</details>

</ul>
</details>

<p>I think it is all bufferutil, but will clean everything and run again....</p>

<p>which gets you to the bit that then requires MSBuild... (all to bundle and process some javascript :-o  )</p>

<p>at which point I shrug and walk away <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>will post logs shortly...</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43897?focusedId=297579&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297579">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I've read that yarn is a little bit smarter about what kind of optional dependencies are built, or that it at least won't fail the entire build if an optional dependency can't be installed. <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=teilo" class="user-hover" rel="teilo">teilo</a> I have a TODO to test this PR on Windows: <a href="https://github.com/jenkinsci/blueocean-plugin/pull/981" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/981</a></p>

<p>Let me give that a test today and let you know what the results are.</p>
</pre>
</details>

</ul>
</details>

<p>I've read that yarn is a little bit smarter about what kind of optional dependencies are built, or that it at least won't fail the entire build if an optional dependency can't be installed. <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=teilo" class="user-hover" rel="teilo">teilo</a> I have a TODO to test this PR on Windows: <a href="https://github.com/jenkinsci/blueocean-plugin/pull/981" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/981</a></p>

<p>Let me give that a test today and let you know what the results are.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43897?focusedId=297690&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297690">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=teilo" class="user-hover" rel="teilo">teilo</a> - any more detail would help us avoid this. </p>


<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> is it reasonable to aim to avoid any dependencies that require a native build component? we often dont' know ahead of time if node-gym will be involved... of course yarn may help here, like you say. </p>
</pre>
</details>

</ul>
</details>

<p>Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=teilo" class="user-hover" rel="teilo">teilo</a> - any more detail would help us avoid this. </p>


<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> is it reasonable to aim to avoid any dependencies that require a native build component? we often dont' know ahead of time if node-gym will be involved... of course yarn may help here, like you say. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/teilo.png" title="teilo's avatar" /> teilo:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43897?focusedId=297844&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297844">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale" class="user-hover" rel="michaelneale">michaelneale</a> Logs from the builds (without python and with python) are attached</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale" class="user-hover" rel="michaelneale">michaelneale</a> Logs from the builds (without python and with python) are attached</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43897?focusedId=297951&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297951">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Yes, bufferutil it seems to be. Not sure why that would really be needed (obviously only a devtime thing). We may be able to exclude it (but then perhaps something else will end up needing a native build). </p>

<p> </p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> is currently quite ill, but when he is better he might get a chance to take a look at this, see if we can exclude it, or if yarn doesn't do this. </p>

<p> </p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=teilo" class="user-hover" rel="teilo">teilo</a> did you get to try the build of <a href="https://github.com/jenkinsci/blueocean-plugin/pull/981" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/981</a> (which uses yarn, not npm) to see if that helps at all? </p>
</pre>
</details>

</ul>
</details>

<p>Yes, bufferutil it seems to be. Not sure why that would really be needed (obviously only a devtime thing). We may be able to exclude it (but then perhaps something else will end up needing a native build). </p>

<p> </p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> is currently quite ill, but when he is better he might get a chance to take a look at this, see if we can exclude it, or if yarn doesn't do this. </p>

<p> </p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=teilo" class="user-hover" rel="teilo">teilo</a> did you get to try the build of <a href="https://github.com/jenkinsci/blueocean-plugin/pull/981" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/981</a> (which uses yarn, not npm) to see if that helps at all? </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43897?focusedId=297953&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297953">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Might get Nicu to take a run at this one at some point.. </p>
</pre>
</details>

</ul>
</details>

<p>Might get Nicu to take a run at this one at some point.. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/teilo.png" title="teilo's avatar" /> teilo:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43897?focusedId=306668&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-306668">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Build (1.1 branch is just DOA on windows as npm is old and buggy....)</p>

<p><a href="https://github.com/nodejs/node-gyp/issues/972" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/nodejs/node-gyp/issues/972</a></p>
</pre>
</details>

</ul>
</details>

<p>Build (1.1 branch is just DOA on windows as npm is old and buggy....)</p>

<p><a href="https://github.com/nodejs/node-gyp/issues/972" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/nodejs/node-gyp/issues/972</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/teilo.png" title="teilo's avatar" /> teilo:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43897?focusedId=306689&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-306689">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>After updating that then browsify barfs  (1.1.0-release branch)</p>
</pre>
</details>

</ul>
</details>

<p>After updating that then browsify barfs  (1.1.0-release branch)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43897?focusedId=307119&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-307119">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>What is actionable about this ticket? It feels like a non-specific bag of problems. Perhaps separate actionable tickets to replace would be a good approach?</p>
</pre>
</details>

</ul>
</details>

<p>What is actionable about this ticket? It feels like a non-specific bag of problems. Perhaps separate actionable tickets to replace would be a good approach?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43897?focusedId=307120&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-307120">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>yes this isn't going to get fixed on the 1.1 branch no matter what anyway, so not sure there is much we can do. We can't have exhaustive control over whatever node library brings in a native depedency at build time anyway, it seems futile to try. </p>
</pre>
</details>

</ul>
</details>

<p>yes this isn't going to get fixed on the 1.1 branch no matter what anyway, so not sure there is much we can do. We can't have exhaustive control over whatever node library brings in a native depedency at build time anyway, it seems futile to try. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43897?focusedId=307122&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-307122">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>So once option is to, as a matter of course, update node/npm versions, libraries etc (I honestly doubt it would help) - but there is no systematic way to know that some library wont' require a native build and tools. I don't think there is anything concrete we can do. if we find out which one, it could well be a different one suddently. Basically with a npm based front end, there is an implicit requirement for a native build environment - which we can't really control. </p>

<p> </p>

<p>Often it is something as "harmless" sounding as Sass, which for some reason needs a C compiler <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/warning.png" height="16" width="16" align="absmiddle" alt="" border="0"/>. Some other options are shouting into the wind, tilting at windmills, or opening tickets with npm/appropriate libraries about its support for windows. </p>
</pre>
</details>

</ul>
</details>

<p>So once option is to, as a matter of course, update node/npm versions, libraries etc (I honestly doubt it would help) - but there is no systematic way to know that some library wont' require a native build and tools. I don't think there is anything concrete we can do. if we find out which one, it could well be a different one suddently. Basically with a npm based front end, there is an implicit requirement for a native build environment - which we can't really control. </p>

<p> </p>

<p>Often it is something as "harmless" sounding as Sass, which for some reason needs a C compiler <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/warning.png" height="16" width="16" align="absmiddle" alt="" border="0"/>. Some other options are shouting into the wind, tilting at windmills, or opening tickets with npm/appropriate libraries about its support for windows. </p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `epic` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-40392%22">JENKINS-40392</a>]</i>
